src/server/protocol.d.ts

```diff
@@ -518,6 +518,11 @@ declare namespace ts.server.protocol {
          * Then the known content will be used upon opening instead of the disk copy   
          */
         fileContent?: string;
+        /**
+         * Used to specify the script kind of the file explicitly. It could be one of the following:
+         *      ".ts", ".js", ".tsx", ".jsx"
+         */
+        scriptKindName?: string;

```

Comment:
i would keep this insync with ScriptKind. so `JS`, `JSX`,  `TS`, `TSX`.

also i would use string literal types, now that we have them instead of `string`. so:

`scriptKindName?: "JS" | "JSX" |  "TS" | "TSX";`